x-pack/plugins/security_solution/public/flyout/document_details/right/components/header_title.tsx

```diff
@@ -5,133 +5,91 @@
-import { CopyToClipboard } from '../../../shared/components/copy_to_clipboard';
-import { useGetAlertDetailsFlyoutLink } from '../../../../timelines/components/side_panel/event_details/use_get_alert_details_flyout_link';
 import { DocumentStatus } from './status';
-import { useAssistant } from '../hooks/use_assistant';
-import {
-  ALERT_SUMMARY_CONVERSATION_ID,
-  EVENT_SUMMARY_CONVERSATION_ID,
-} from '../../../../common/components/event_details/translations';
 import { DocumentSeverity } from './severity';
 import { RiskScore } from './risk_score';
 import { useBasicDataFromDetailsData } from '../../../../timelines/components/side_panel/event_details/helpers';
 import { useRightPanelContext } from '../context';
 import { PreferenceFormattedDate } from '../../../../common/components/formatted_date';
-import { FLYOUT_HEADER_TITLE_TEST_ID, SHARE_BUTTON_TEST_ID } from './test_ids';
-
-export interface HeaderTitleProps {
-  /**
-   * If false, update the margin-top to compensate the fact that the expand detail button is not displayed
-   */
-  flyoutIsExpandable: boolean;
-}
+import { RenderRuleName } from '../../../../timelines/components/timeline/body/renderers/formatted_field_helpers';
+import { SIGNAL_RULE_NAME_FIELD_NAME } from '../../../../timelines/components/timeline/body/renderers/constants';
+import { FLYOUT_HEADER_TITLE_TEST_ID } from './test_ids';
 
 /**
  * Document details flyout right section header
  */
-export const HeaderTitle: VFC<HeaderTitleProps> = memo(({ flyoutIsExpandable }) => {
-  const { dataFormattedForFieldBrowser, eventId, indexName } = useRightPanelContext();
-  const { isAlert, ruleName, timestamp } = useBasicDataFromDetailsData(
+export const HeaderTitle: FC = memo(() => {
+  const { dataFormattedForFieldBrowser, eventId, scopeId } = useRightPanelContext();
+  const { isAlert, ruleName, timestamp, ruleId } = useBasicDataFromDetailsData(
     dataFormattedForFieldBrowser
   );
-  const alertDetailsLink = useGetAlertDetailsFlyoutLink({
-    _id: eventId,
-    _index: indexName,
-    timestamp,
-  });
 
-  const showShareAlertButton = isAlert && alertDetailsLink;
+  const { euiTheme } = useEuiTheme();
 
-  const { showAssistant, promptContextId } = useAssistant({
-    dataFormattedForFieldBrowser,
-    isAlert,
-  });
+  const ruleTitle = useMemo(
+    () => (
+      <RenderRuleName
+        contextId={scopeId}
+        eventId={eventId}
+        fieldName={SIGNAL_RULE_NAME_FIELD_NAME}
+        fieldType={'string'}
+        isAggregatable={false}
+        isDraggable={false}
+        linkValue={ruleId}
+        value={ruleName}
+        openInNewTab
+      >
+        <div>
+          <EuiIcon type={'warning'} size="m" className="eui-alignBaseline" />
+          &nbsp;
+          <EuiTitle size="s">
+            <EuiTextColor color={euiTheme.colors.primaryText}>
+              <span data-test-subj={FLYOUT_HEADER_TITLE_TEST_ID}>{ruleName}</span>
```

Comment:
@christineweng see Fernanda's [comment](https://github.com/elastic/kibana/pull/170279#issuecomment-1790925199) :)

Reply:
The line clamp plus the icon is trickier than i thought.. An issue I'm having is that the end of the last line + the eclipses does not always extends to the right. it would look like there is a gap between the text and the popout icon.

I updated the line clamp but removed the icon in this PR, because it doesn't look good at all. I want to tackle this in a separate PR and extract the title into its own component with story book, wdyt? 

![image](https://github.com/elastic/kibana/assets/18648970/2c00916d-e58f-4fae-8978-2f722f608c95)


https://github.com/elastic/kibana/assets/18648970/bd39fade-c5a7-4786-8ea9-638cab030a50

Reply:
yeah I see how this could be a bit tricky.. I'm fine taking care of it in a separate PR!

Reply:
but honestly, looking at your video, the gap doesn't bother me that much. The icon is always right-aligned and that looks good to me!